scripts/processDiagnosticMessages.mjs

```diff
@@ -85,27 +85,25 @@ function buildInfoFileOutput(messageTable, inputFilePathRel) {
     const result = [
         "// <auto-generated />",
         `// generated from '${inputFilePathRel}'`,
+        "/* eslint-disable @typescript-eslint/naming-convention */",
         "",
         "import { DiagnosticCategory, DiagnosticMessage } from \"./types\";",
         "",
         "function diag(code: number, category: DiagnosticCategory, key: string, message: string, reportsUnnecessary?: {}, elidedInCompatabilityPyramid?: boolean, reportsDeprecated?: {}): DiagnosticMessage {",
         "    return { code, category, key, message, reportsUnnecessary, elidedInCompatabilityPyramid, reportsDeprecated };",
         "}",
         "",
-        "/** @internal */",
-        "export const Diagnostics = {",
     ];
     messageTable.forEach(({ code, category, reportsUnnecessary, elidedInCompatabilityPyramid, reportsDeprecated }, name) => {
         const propName = convertPropertyName(name);
         const argReportsUnnecessary = reportsUnnecessary ? `, /*reportsUnnecessary*/ ${reportsUnnecessary}` : "";
         const argElidedInCompatabilityPyramid = elidedInCompatabilityPyramid ? `${!reportsUnnecessary ? ", /*reportsUnnecessary*/ undefined" : ""}, /*elidedInCompatabilityPyramid*/ ${elidedInCompatabilityPyramid}` : "";
         const argReportsDeprecated = reportsDeprecated ? `${!argElidedInCompatabilityPyramid ? ", /*reportsUnnecessary*/ undefined, /*elidedInCompatabilityPyramid*/ undefined" : ""}, /*reportsDeprecated*/ ${reportsDeprecated}` : "";
 
-        result.push(`    ${propName}: diag(${code}, DiagnosticCategory.${category}, "${createKey(propName, code)}", ${JSON.stringify(name)}${argReportsUnnecessary}${argElidedInCompatabilityPyramid}${argReportsDeprecated}),`);
+        // TODO(jakebailey): /* @__PURE__ */ should work here, but doesn't.
+        result.push(`/** @internal */ export const ${propName} = diag(${code}, DiagnosticCategory.${category}, "${createKey(propName, code)}", ${JSON.stringify(name)}${argReportsUnnecessary}${argElidedInCompatabilityPyramid}${argReportsDeprecated});`);

```

Comment:
Did you want to use `@__PURE__` because of the call to `diag`? Could you instead just create the object literals directly?

Reply:
Yes, it was due to the call; I was trying to get esbuild to remove all of the diagnostics that aren't needed but they weren't, and my assumption is that because it's a function call, it doesn't. But, even though the comment is present, it doesn't delete them, so that's confusing.

I'll give literals a try instead (I was just avoiding it for code size reasons).

Reply:
Yeah, unfortunately, doesn't help:

![image](https://user-images.githubusercontent.com/5341706/211902024-72e01ddf-8690-4730-81f6-8278cb78d678.png)

Must be some bug in esbuild's tree shaking or something.